services/repository/src/test/java/org/sagebionetworks/repo/web/filter/UserThrottleFilterTest.java

```diff
@@ -1,102 +1,161 @@
-		request.setParameter(AuthorizationConstants.USER_ID_PARAM, "111");
-		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-
-		when(userThrottleGate.attemptToAcquireLock("111", LOCK_TIMOUTE_SEC, MAX_CONCURRENT_LOCKS)).thenThrow(new RuntimeException());
+	public void testConcurrentConnectionsAcquireLockException() throws Exception {
+		when(userThrottleGate.attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS)).thenThrow(new RuntimeException());
 		try {
 			filter.doFilter(request, response, filterChain);
 		} finally {
-			verify(userThrottleGate).attemptToAcquireLock("111", LOCK_TIMOUTE_SEC, MAX_CONCURRENT_LOCKS);
-			verifyNoMoreInteractions(filterChain, userThrottleGate);
+			verify(userThrottleGate).attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS);
+			verifyNoMoreInteractions(filterChain, userThrottleGate, userFrequencyThrottleGate);
+		}
+	}
+	
+	@Test(expected = ServletException.class)
+	public void testRequestFrequencyAcquireLockException() throws Exception {
+		when(userFrequencyThrottleGate.attemptToAcquireLock(userId, REQUEST_FREQUENCY_LOCK_TIMEOUT_SEC, MAX_REQUEST_FREQUENCY_LOCKS)).thenThrow(new RuntimeException());
+		try {
+			filter.doFilter(request, response, filterChain);
+		} finally {
+			verify(userThrottleGate).attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS);
+			verifyNoMoreInteractions(filterChain, userThrottleGate, userFrequencyThrottleGate);
 		}
 	}
 
 	@Test
-	public void testNoEmptySlots() throws Exception {
-		MockHttpServletRequest request = new MockHttpServletRequest();
-		request.setParameter(AuthorizationConstants.USER_ID_PARAM, "111");
-		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
+	public void testNoEmptyConcurrentConnectionSlots() throws Exception {
 		Consumer consumer = mock(Consumer.class);
 		ReflectionTestUtils.setField(filter, "consumer", consumer);
 
-		when(userThrottleGate.attemptToAcquireLock("111", LOCK_TIMOUTE_SEC, MAX_CONCURRENT_LOCKS)).thenReturn(null);
+		when(userThrottleGate.attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS)).thenReturn(null);
+		when(userFrequencyThrottleGate.attemptToAcquireLock(userId, REQUEST_FREQUENCY_LOCK_TIMEOUT_SEC, MAX_REQUEST_FREQUENCY_LOCKS)).thenReturn(true);
 
+		
 		filter.doFilter(request, response, filterChain);
-		assertEquals(503, response.getStatus());
+		assertEquals(HttpStatus.SC_SERVICE_UNAVAILABLE, response.getStatus());
+		
+		ArgumentCaptor<ProfileData> profileDataArgument = ArgumentCaptor.forClass(ProfileData.class); 
+		verify(consumer).addProfileData(profileDataArgument.capture());
+		assertEquals("ConcurrentConnectionsLockUnavailable", profileDataArgument.getValue().getName());
```

Comment:
make ConcurrentConnectionsLockUnavailable a constant in UserThrottleFilter and refer to it here